mne/encoding/feature.py

```diff
@@ -0,0 +1,157 @@
+
+        Parameters
+        ----------
+        event_ixs : array, shape (n_events)
+            The event indices or onsets in seconds. `If self.sfreq == 1`,
+            then these are assumed to be indices. If `self.sfreq != 1`, these
+            should be onsets in seconds and will be multiplied by `sfreq`.
+        event_ids : array-like, shape (n_events) | None
+            The event ID for each event. If None, all events are assumed to be
+            of same type.
+        event_dict : dict
+            A dictionary of (event_id: event_id_name) pairs. Defines a string
+            name for each event type.
+        """
+        if event_ixs.ndim > 1:
+            raise ValueError("events must be shape (n_trials, 3),"
+                             " found shape %s" % event_ixs.shape)
+        event_ids = np.ones_like(event_ixs) if event_ids is None else event_ids
+        unique_event_types = np.unique(event_ids)
+        if event_dict is None:
+            event_dict = dict((ii, 'event_%s' % ii)
+                              for ii in unique_event_types)
+
+        # Turn event_ixs from seconds to indices in case sfreq != 1
+        event_ixs = event_ixs * self.sfreq
+        event_ixs = event_ixs.astype(int)
+
+        # Iterate through event types and create columns of event onsets.
+        events_continuous = np.zeros([len(unique_event_types), self.n_times])
+        event_names = []
+        for ii, ev_type in enumerate(unique_event_types):
+            msk_events = event_ids == ev_type
+            i_ev = event_ixs[msk_events]
+            events_continuous[ii, i_ev] = 1
+
+            # Handle event names
+            event_names.append(event_dict[ev_type])
+        self.event_names = event_names
+        self.events_continuous = events_continuous
+        self.unique_event_types = unique_event_types
+
+    def transform(self, X=None, y=None):
+        """Return continuous events data.
+
+        Returns
+        -------
+        events_continuous : array, shape (n_unique_events, n_times)
+            A binary array with one row for each unique event type. 1s
+            in the array correspond to event onsets for that row.
+        """
+        return self.events_continuous
+
+    def fit_transform(self, event_ixs, event_ids=None, event_dict=None):
+        self.fit(event_ixs, event_ids=event_ids, event_dict=event_dict)
+        return self.transform()
+
+
+class DataSubsetter(object):
+    """Return a subset of data."""
+    def __init__(self, ixs):
+        self.ixs = ixs.astype(int)
+
+    def fit(self, data, y=None):
+        if data.shape[-1] < self.ixs.max():
+            raise ValueError('ixs exceed data shape')
+        self.data = np.asarray(data)
+
+    def transform(self, X=None, y=None):
+        return self.data[..., self.ixs]
+
+    def fit_transform(self, ixs, y=None):
+        self.fit(ixs)
+        return self.transform()
+
+
+def delay_timeseries(ts, sfreq, delays):
```

Comment:
Cool - I'll make it it private. I think I missed some of the comments on the last iteration because of moving this to the new folder